druid/src/main/java/org/apache/calcite/adapter/druid/DruidConnectionImpl.java

```diff
@@ -350,24 +350,24 @@ private void parseField(List<String> fieldNames, List<ColumnMetaData.Rep> fieldT
           if (s.equals("Infinity") || s.equals("-Infinity") || s.equals("NaN")) {
             throw new RuntimeException("/ by zero");
           }
+          rowBuilder.set(i, s);
+          break;
         case FLOAT:
         case PRIMITIVE_FLOAT:
         case PRIMITIVE_DOUBLE:
         case NUMBER:
         case DOUBLE:
           if (s.equals("Infinity")) {
             rowBuilder.set(i, Double.POSITIVE_INFINITY);
-            break;
           } else if (s.equals("-Infinity")) {
             rowBuilder.set(i, Double.NEGATIVE_INFINITY);
-            break;
           } else if (s.equals("NaN")) {
             rowBuilder.set(i, Double.NaN);
-            break;
+          } else {
+            rowBuilder.set(i, s);
           }
-          //fallthrough
+          break;
         default:
-          rowBuilder.set(i, s);
```

Comment:
Why did you remove this line "rowBuilder.set(i, s);"? We still want to set the value if the type is not numeric.